buildSrc/src/main/java/UpdateContributorStatisticsTask.kt

```diff
@@ -0,0 +1,175 @@
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.net.URL
+
+/** Counts the changes made by each author from the git log, merges multiple authors that are linked
+ *  to the same github account and writes the result into a JSON file. */
+open class UpdateContributorStatisticsTask : DefaultTask() {
+    @get:Input lateinit var targetFile: String
+    @get:Input var skipCommits: Set<String> = setOf()
+    @get:Input var skipCommitRegex: Regex? = null
+    @get:Input lateinit var codeFileRegex: Regex
+    @get:Input lateinit var assetFileRegex: Regex
+    @get:Input lateinit var interfaceMarkupRegex: Regex
+    @get:Input lateinit var githubApiToken: String
+
+    @TaskAction fun run() {
+        val contributors = getContributors()
+
+        for (contributor in contributors) {
+            val githubUser = downloadGithubUserDetails(contributor.hash!!)
+            contributor.githubUsername = githubUser?.login
+        }
+
+        val mergedContributors = mergeContributorsByGithubAccount(contributors)
+            .filter { it.linesOfCodeChanged + it.assetFilesChanged + it.linesOfInterfaceMarkupChanged > 0 }
+
+        writeContributors(mergedContributors.sortedByDescending { it.linesOfCodeChanged })
+    }
+
+    private fun getContributors(): Collection<Contributor> {
+        val result = mutableMapOf<String, Contributor>()
+        val assetFilesChangedByCommit = mutableMapOf<String, Int>()
+        val linesOfInterfaceMarkupChangedByCommit = mutableMapOf<String, Int>()
+        val linesOfCodeChangedByCommit = mutableMapOf<String, Int>()
+        val ignoredCommits = mutableSetOf<String>()
+        val command = "git log --pretty=\"name: %an%nhash: %H%nsubject: %s%ntimestamp: %at\" --numstat"
```

Comment:
```suggestion
        val command = "git log --pretty=\"name: %an%nhash: %H%nsubject: %s%ntimestamp: %at\" --numstat --no-merges"
```

`--no-merges` was [included](https://github.com/streetcomplete/StreetComplete/pull/2815/files#diff-4b7de3cd0f59bb71c8340c4c9b7c4051771476742cd9b0eb9ea2cbf570f5b9caR22) in #2815, and this explicitly excluded commit suggests that merge commits should indeed be excluded:

https://github.com/streetcomplete/StreetComplete/blob/fa1ca0a3ab90aecdf519434a1d6c62c560709b3f/build.gradle.kts#L65

Reply:
Yeah, but for some reason the commit you referred to is not marked as a merge. I tested both with --no-merges and without and there was almost no difference IIRC.

Reply:
Actually with --no-merges, it is actually more (a tiny amount), so that option obviously doesn't do what we think it does.

Reply:
This is the output of the 20 most recent commits on `master`, with and without `--no-merges`:

```diff
- git log --oneline | head -n 20
+ git log --oneline --no-merges | head -n 20

  96f24b03a update changelog and version up
  ada68534a update changelog
  c4eb4ec79 update changelog
  1dc3e25f3 horizontal measuring: require first and second point to be on the same plane (#3832)
  f6ac02097 fix #3843
  c12dfa83b fix #3843
  0a2459475 Passage between two barriers -> Passage between barriers (fixes #3852)
- 55fe84498 Merge pull request #3758 from fatal69100/add_seating_quest
  f29253013 small code style changes
  7b2e17117 replace seating icon
  0d342e809 Rename icon to match SVG file name
  68fd60054 Lint
  eae6bfa3e typo
  3d15dedf9 add allotments(=plot) to is-area expression
  fdb64d616 update changelog for the next version
- c662b34af Merge pull request #3846 from mnalis/barrier-not-cyclingbarrier
  125f70acc shorten code a bit
- 793e52724 Merge pull request #3845 from mnalis/barrier-not-bollard
  d4b97e4d7 shorten code a bit
- 9e4d8c12c Merge pull request #3844 from arrival-spring/incorrect-tactile
+ 42b5b5d81 assume some roads to be paved
+ e70e72384 add OtherAnswer "not a cycle barrier"
+ 6e80c772e Rename to TactilePavingCrosswalkAnswer
+ fb95f68ac missing import
```

So I think that CLI parameter works correctly and thus should be added. I know that ae7a244dd60ccfc91cf2dc01bf9e60c8d6a81616 is not excluded for some reason (that's why it's in the skip list), but it seems to be an exception.

Reply:
> Actually with --no-merges, it is actually more (a tiny amount)

More what?

Reply:
More contributions in the resulting JSON file